addon/session-stores/cookie.js

```diff
@@ -76,25 +77,51 @@ export default BaseStore.extend({
   */
   cookieExpirationTime: null,
 
-  _secureCookies: window.location.protocol === 'https:',
+  _cookies: computed(function() {
+    let owner = getOwner(this);
+
+    return owner.lookup('service:cookies');
+  }),
+
+  _fastboot: computed(function() {
+    let owner = getOwner(this);
+
+    return owner.lookup('service:fastboot');
+  }),
+
+  _secureCookies: computed(function() {
+    if (this.get('_fastboot.isFastBoot')) {
+      return this.get('_fastboot._fastbootInfo.request.hostname').indexOf('https:') === 0;
+    } else {
+      return window.location.protocol === 'https:'
+    }
+  }).volatile(),
 
   _syncDataTimeout: null,
 
   _renewExpirationTimeout: null,
 
   _isPageVisible: computed(function() {
-    const visibilityState = document.visibilityState || 'visible';
-    return visibilityState === 'visible';
+    if (this.get('_fastboot.isFastBoot')) {
+      return false;
+    } else {
+      const visibilityState = typeof document !== 'undefined' ? document.visibilityState || 'visible' : false;
+      return visibilityState === 'visible';
+    }
   }).volatile(),
 
   init() {
     this._super(...arguments);
-
-    next(() => {
-      this._syncData().then(() => {
-        this._renewExpiration();
+/*
+    if (!this.get('_fastboot.isFastBoot')) {
+      next(() => {
+        this._syncData().then(() => {
+          this._renewExpiration();
+        });
       });
-    });
+    } else {
+      this._renew();
+    }*/
   },
```

Comment:
Oh, that was me ;)

I guess I did that probably only because cookie access wasn't working in FastBoot back then. Should be good to uncomment that now.